Parse/Parse.m

```diff
@@ -54,17 +53,25 @@ + (void)initialize {
 ///--------------------------------------
 
 + (void)setApplicationId:(NSString *)applicationId clientKey:(NSString *)clientKey {
-    PFConsistencyAssert([applicationId length], @"'applicationId' should not be nil.");
-    PFConsistencyAssert([clientKey length], @"'clientKey' should not be nil.");
-    
-    // Setup new manager first, so it's 100% ready whenever someone sends a request for anything.
-    ParseManager *manager = [[ParseManager alloc] initWithApplicationId:applicationId clientKey:clientKey];
-    [manager configureWithApplicationGroupIdentifier:applicationGroupIdentifier_
-                     containingApplicationIdentifier:containingApplicationBundleIdentifier_
-                               enabledLocalDataStore:shouldEnableLocalDatastore_];
-    currentParseManager_ = manager;
+    currentParseConfiguration_.applicationId = applicationId;
+    currentParseConfiguration_.clientKey = clientKey;
+
+    [self initializeWithConfiguration:currentParseConfiguration_];
+
+    // TODO(richardross): Investigate why this is necessary.
+    currentParseConfiguration_.localDatastoreEnabled = NO;
+}
 
-    shouldEnableLocalDatastore_ = NO;
++ (void)initializeWithConfiguration:(ParseClientConfiguration *)configuration {
```

Comment:
This should have a validation on `applicationId`, `clientKey` not being `nil`, or rather not having 0 length.